kernel/src/process/posix_thread/mod.rs

```diff
@@ -171,10 +174,32 @@ impl PosixThread {
         return_errno_with_message!(Errno::EPERM, "sending signal to the thread is not allowed.");
     }
 
+    /// Sets the input [`Waker`] as the signal waker of this thread or
+    /// clears the signalled waker if the input is `None`.
+    ///
+    /// This approach can collaborate with signal-aware wait methods.
+    /// Once a signalled waker is set for a thread, it cannot be reset until it is cleared.
+    ///
+    /// # Panics
+    ///
+    /// If setting a new waker before clearing the current thread's signalled waker
+    /// this method will panic.
+    pub fn set_signalled_waker(&self, waker: Option<Arc<Waker>>) {
+        let mut signalled_waker = self.signalled_waker.lock();
+        if signalled_waker.is_some() && waker.is_some() {
+            panic!();
+        }
+
```

Comment:
- Use `assert!()`.
 - Prefer to have two methods, `set_signnalled_waker` and `clear_signalled_waker`.
 - Remove this blank line.